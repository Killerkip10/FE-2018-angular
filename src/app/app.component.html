<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
