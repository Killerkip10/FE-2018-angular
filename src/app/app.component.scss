app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
