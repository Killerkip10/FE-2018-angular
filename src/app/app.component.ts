import {Component, OnInit} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';

import {Store} from '@ngrx/store';
import {GetProfile} from './store/actions/profile';

import {TokenService} from './core/services';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {
  public tokenService = TokenService;

  constructor(
    translateService: TranslateService,
    private store: Store<null>
  ) {
    translateService.addLangs(['en', 'ru']);
    translateService.setDefaultLang('en');
  }

  ngOnInit(): void {
    this.store.dispatch(new GetProfile());
  }
}
