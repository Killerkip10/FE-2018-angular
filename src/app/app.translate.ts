import {NgModule} from '@angular/core';
import {TranslateModule, TranslateLoader} from '@ngx-translate/core';
import {TranslateHttpLoader} from '@ngx-translate/http-loader';
import {HttpClient} from '@angular/common/http';

import {urlConfig} from './config';

@NgModule({
  imports: [
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: HttpLoaderFactory,
        deps: [HttpClient]
      }
    })
  ]
})
export class AppTranslate {}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(http, `${urlConfig.url}i18n/`, '.json');
}
