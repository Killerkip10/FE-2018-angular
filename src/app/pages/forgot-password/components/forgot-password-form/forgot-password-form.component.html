<section>
  <mat-card>
    <mat-toolbar color="primary">{{'FORGOT-PASSWORD' | translate}}</mat-toolbar>

    <mat-card-content>
      <form #forgotForm="ngForm" (submit)="submit(forgotForm)">
        <div>
          <mat-form-field>
            <input
              matInput
              required
              ngModel
              name="login"
              type="text"
              [placeholder]="'LOGIN' | translate">
            <mat-icon matSuffix>account_circle</mat-icon>
          </mat-form-field>

          <mat-form-field>
            <input
              matInput
              [ngModel]="password"
              disabled
              type="text"
              name="password"
              [placeholder]="'PASSWORD' | translate">
            <mat-icon matSuffix>lock</mat-icon>
          </mat-form-field>
        </div>

        <p class="err-message" *ngIf="errorMessage">{{errorMessage | translate}}</p>

        <div class="actions">
          <button mat-flat-button [disabled]="!forgotForm.valid" color="primary">{{'SEND' | translate}}</button>

          <a mat-button [routerLink]="['/login']">{{'SIGN-IN' | translate}}</a>
        </div>
      </form>
    </mat-card-content>
  </mat-card>
</section>
