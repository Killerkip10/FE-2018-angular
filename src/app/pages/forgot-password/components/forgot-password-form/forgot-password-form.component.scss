mat-card{
  padding: 0;
  mat-card-content{
    padding: 16px;
    .err-message{
      margin-top: 0;
      color: red;
    }
    mat-form-field{
      width: 100%;
    }
    .actions{
      display: flex;
      justify-content: space-between;
    }
  }
}
