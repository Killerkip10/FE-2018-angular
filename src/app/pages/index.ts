export * from './user-profile/user-profile-page.module';
export * from './login/login-page.module';
export * from './forgot-password/forgot-password-page.module';
export * from './user-list/user-list.module';
