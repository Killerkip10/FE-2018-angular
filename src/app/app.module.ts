import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {RouterModule} from '@angular/router';
import {TranslateModule} from '@ngx-translate/core';

import {CoreModule} from './core/core.module';
import {AppRouting} from './app.routing';
import {AppComponent} from './app.component';
import {AppTranslate} from './app.translate';
import {
  MatSidenavModule,
  MatListModule
} from '@angular/material';
import {
  LoginPageModule,
  UserProfilePageModule,
  ForgotPasswordPageModule,
  SearchUsersPageModule
} from './pages';


@NgModule({
  imports: [
    CoreModule,
    CommonModule,
    RouterModule,
    TranslateModule,

    LoginPageModule,
    UserProfilePageModule,
    ForgotPasswordPageModule,
    SearchUsersPageModule,

    MatSidenavModule,
    MatListModule,

    AppTranslate,
    AppRouting
  ],
  declarations: [
    AppComponent
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
