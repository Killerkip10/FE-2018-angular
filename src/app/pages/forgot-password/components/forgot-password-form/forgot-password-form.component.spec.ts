import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {APP_BASE_HREF, CommonModule} from '@angular/common';
import {FormsModule} from '@angular/forms';
import {TranslateModule} from '@ngx-translate/core';
import {RouterTestingModule} from '@angular/router/testing';

import {ForgotPasswordFormComponent} from './forgot-password-form.component';
import {AppTranslate} from '../../../../app.translate';
import {CoreModule} from '../../../../core/core.module';

describe('ForgotPasswordComponent', () => {
  let component: ForgotPasswordFormComponent;
  let fixture: ComponentFixture<ForgotPasswordFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CoreModule,
        CommonModule,
        TranslateModule,
        FormsModule,
        RouterTestingModule,

        AppTranslate
      ],
      declarations: [
        ForgotPasswordFormComponent
      ],
      providers: [
        {provide: APP_BASE_HREF, useValue: '/'}
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ForgotPasswordFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.errorMessage = 'ERROR-TEST';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.password = 'PASSWORD';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
