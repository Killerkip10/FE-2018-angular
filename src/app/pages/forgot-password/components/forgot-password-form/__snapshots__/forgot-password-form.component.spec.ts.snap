// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordComponent snapshot 1`] = `
<app-forgot-password
  authService={[Function AuthService]}
  errorMessage=""
  password=""
>
  <section>
    <mat-card>
      <mat-toolbar
        color="primary"
      >
        FORGOT-PASSWORD
      </mat-toolbar>
      <mat-card-content>
        <form
          class="ng-untouched ng-pristine ng-valid"
          novalidate=""
        >
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-valid"
                matinput=""
                name="login"
                ng-reflect-model=""
                ng-reflect-name="login"
                ng-reflect-required=""
                ngmodel=""
                placeholder="LOGIN"
                required=""
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                account_circle
              </mat-icon>
            </mat-form-field>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-valid"
                disabled=""
                matinput=""
                name="password"
                ng-reflect-is-disabled=""
                ng-reflect-model=""
                ng-reflect-name="password"
                placeholder="PASSWORD"
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                lock
              </mat-icon>
            </mat-form-field>
          </div>
          
          <div
            class="actions"
          >
            <button
              color="primary"
              mat-flat-button=""
            >
              SEND
            </button>
            <a
              href="/login"
              mat-button=""
              ng-reflect-router-link="/login"
            >
              SIGN-IN
            </a>
          </div>
        </form>
      </mat-card-content>
    </mat-card>
  </section>
</app-forgot-password>
`;

exports[`ForgotPasswordComponent snapshot 2`] = `
<app-forgot-password
  authService={[Function AuthService]}
  errorMessage={[Function String]}
  password=""
>
  <section>
    <mat-card>
      <mat-toolbar
        color="primary"
      >
        FORGOT-PASSWORD
      </mat-toolbar>
      <mat-card-content>
        <form
          class="ng-untouched ng-pristine ng-invalid"
          novalidate=""
        >
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-invalid"
                matinput=""
                name="login"
                ng-reflect-model=""
                ng-reflect-name="login"
                ng-reflect-required=""
                ngmodel=""
                placeholder="LOGIN"
                required=""
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                account_circle
              </mat-icon>
            </mat-form-field>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine"
                disabled=""
                matinput=""
                name="password"
                ng-reflect-is-disabled=""
                ng-reflect-model=""
                ng-reflect-name="password"
                placeholder="PASSWORD"
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                lock
              </mat-icon>
            </mat-form-field>
          </div>
          
          <p
            class="err-message"
          >
            ERROR-TEST
          </p>
          <div
            class="actions"
          >
            <button
              color="primary"
              disabled=""
              mat-flat-button=""
            >
              SEND
            </button>
            <a
              href="/login"
              mat-button=""
              ng-reflect-router-link="/login"
            >
              SIGN-IN
            </a>
          </div>
        </form>
      </mat-card-content>
    </mat-card>
  </section>
</app-forgot-password>
`;

exports[`ForgotPasswordComponent snapshot 3`] = `
<app-forgot-password
  authService={[Function AuthService]}
  errorMessage=""
  password={[Function String]}
>
  <section>
    <mat-card>
      <mat-toolbar
        color="primary"
      >
        FORGOT-PASSWORD
      </mat-toolbar>
      <mat-card-content>
        <form
          class="ng-untouched ng-pristine ng-invalid"
          novalidate=""
        >
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-invalid"
                matinput=""
                name="login"
                ng-reflect-model=""
                ng-reflect-name="login"
                ng-reflect-required=""
                ngmodel=""
                placeholder="LOGIN"
                required=""
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                account_circle
              </mat-icon>
            </mat-form-field>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine"
                disabled=""
                matinput=""
                name="password"
                ng-reflect-is-disabled=""
                ng-reflect-model="PASSWORD"
                ng-reflect-name="password"
                placeholder="PASSWORD"
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                lock
              </mat-icon>
            </mat-form-field>
          </div>
          
          <div
            class="actions"
          >
            <button
              color="primary"
              disabled=""
              mat-flat-button=""
            >
              SEND
            </button>
            <a
              href="/login"
              mat-button=""
              ng-reflect-router-link="/login"
            >
              SIGN-IN
            </a>
          </div>
        </form>
      </mat-card-content>
    </mat-card>
  </section>
</app-forgot-password>
`;
